pkg/objstore/objstore.go

```diff
@@ -145,6 +146,12 @@ func TryToGetSize(r io.Reader) (int64, error) {
 		return int64(f.Len()), nil
 	case *strings.Reader:
 		return f.Size(), nil
+	case *minio.Object:
```

Comment:
💪🏻   Could add a unit test?

Reply:
pkg/objstore is a abstract package for all of objectstore, i think it should not assert to a certain implement. another way to solve the replicate issue is implement `objstore.ObjectSizer` for this . like the PR #4687.
"[pkg/objstore](https://github.com/thanos-io/thanos/pull/4687/files/026a4b7276cbb4fbb5d41bc454c799f388ee8210#diff-41e5eeb71bb7097b6a59cdc9eb0002efd60df6b42394773d6d444eb8fec80676)"

```
	// Add size info into reader for pass it to Upload function.
	r := objectSizerReadCloser{ReadCloser: resp.Body, size: resp.ContentLength}
	return r, nil
}

type objectSizerReadCloser struct {
	io.ReadCloser
	size int64
}

// ObjectSize implement objstore.ObjectSizer.
func (o objectSizerReadCloser) ObjectSize() (int64, error) {
	return o.size, nil
}
```

Reply:
Makes sense. Thank you!